lib/sync.js

```diff
@@ -62,9 +89,15 @@ function Sync(src, target, opts, callback) {
     return function (event, func) {
       var isChange = event === 'change' &&
         (func === pullChange || func === pushChange);
+      var isDenied = event === 'denied' &&
+        (func === pullDenied || func === pushDenied);
```

Comment:
I don't believe this `denied` stuff is tested, although it's undoubtedly necessary. Would be nice to add a test to confirm this is the right behavior (you can run `COVERAGE=1 npm t` to verify that this code is being hit).